src/osd/OSD.cc

```diff
@@ -3102,6 +3102,19 @@ will start to track new ops received afterwards.";
       f, false, logger, counter);
   }
 
+  else if (prefix == "trim stale osdmaps") {
+    epoch_t last_epoch = 0;
+    while (last_epoch != superblock.oldest_map) {
+      lock_guard l(osd_lock);
+      int ret = trim_stale_maps(last_epoch);
+      if (ret) {
+        dout(20) << "Trimmed " << ret <<  " maps" << dendl;
+      } else {
+        dout(20) << "No stale osdmaps found" << dendl;
```

Comment:
I'd consider adding a break here. I know that we should never have 0 returned unless we are at oldest_map - but it would simplify validation.

Reply:
Originally we shouldn't break here. The intention was to release the lock every n iterations. Even if no stale osdmaps were found in the last iteration, we would have released the lock the retake it to continue to the next iteration. 
However, this is no longer relevant since no lock is required, the while loop is removed.